pkg/roachpb/api.proto

```diff
@@ -402,18 +402,34 @@ message ClearRangeResponse {
 }
 
 
-// A RevertRangeRequest specifies a range of keys in which to clear all MVCC
-// revisions more recent than some TargetTime from the underlying engine, thus
-// reverting the range (from the perspective of an MVCC scan) to that time.
+// A RevertRangeRequest specifies a range of keys to revert to some past time.
+// By default, it will clear all revision more recent that TargetTime from the
+// underlying engine. However, this violates several guarantees including MVCC
+// immutability, the closed timestamp, timestamp cache, and others. See the
+// ExperimentalPreserveHistory parameter which will uphold these guarantees.
 message RevertRangeRequest {
   RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 
-  // TargetTime specifies a the time to which to "revert" the range by clearing
-  // any MVCC key with a strictly higher timestamp. TargetTime must be higher
+  // TargetTime specifies a the time to which to "revert" the range to. Any
+  // versions later than TargetTime will be undone. TargetTime must be higher
   // than the GC Threshold for the replica - so that it is assured that the keys
   // for that time are still there — or the request will fail.
   util.hlc.Timestamp target_time = 2 [(gogoproto.nullable) = false];
 
+  // ExperimentalPreserveHistory will preserve MVCC history by, rather than
```

Comment:
We _could_ decouple these, and say 1) PreserveHistory will do the tombstone/undo, and is thus mvcc and 2) experimental_mvcc_delrange allows preserve history, if in use, to use cheap range deletion tombstones rather than slow/expensive individual tombstones. Then we can do mvcc but just slow without being in experimental territory?

Reply:
We could, but that would require more testing and such, and I can't afford that right now. Any reason to believe we would use this in 22.1?

Reply:
nope, works for me.